### ISSUE TITLE
Option to filter lambda names, cloudwatch log groups, ... using a search pattern

### ISSUE THREAD

**Issue by oscaretu** [aws/aws-toolkit-vscode#1877]
**Is your feature request related to a problem? Please describe.**

When there are lots lots of lambdas o cloudwatch log groups, It's a pain to have to keep scrolling and scrolling until you get to the entry you need. In many cases, I work with a specific set of lambdas (we call them a lambda family, and all of them share a prefix). I would like to have an option to reduce the time I need to locate them.

**Describe the solution you'd like**

If there were a input box where I can enter a search pattern so only the lambda names that match that pattern appear in the listing, it would be easier to locate the lambda I want to find among several hundreds that appear now. And the same for CloudWatch log groups. If the pattern were a regex, it would allow a more flexible solution. And the pattern should be resistant to restarts.

**Comment by JadenSimon** [comment]
> And the pattern should be resistant to restarts

Good idea! I'm wondering how we should expose this filtering functionality. Do you think a right-click option on the CloudWatch logs node would make sense? The filtering would then be region-specific.

**Comment by oscaretu** [comment]
@JadenSimon, I think that the option you suggest (right click button) can be OK . For people that has to work in different regions, it would be good be able to store different patterns for different regions. I don't know how difficult it would be to allow patterns to be specified by editing a JSON file, with a hierarchy of regions, and of services within each region, and of patterns within each service.

**Comment by JadenSimon** [comment]
I was thinking for every top-level node in every region, you would be able to add a filter via an input box which is stored in the user's settings. Perhaps a command to open the settings file directly too. 

There are some UX considerations we need to make as well. For example, how should the filter be displayed to the user (maybe appended to the node's name?)

**Comment by oscaretu** [comment]
@JadenSimon, the two options you suggestedfor entering pattern are OK for me. 

One possibility for the UX could be an icon that represents that there is a filter applied to the listing of lambdas, Cloudwatch logs, ... And when you hover on it, is shows a layer with the information about the filtering pattern applied to that service. And if you click on in, you edit it. Or two icons, one for input text and another for opening file with user's settings.

**Comment by JadenSimon** [comment]
Yeah I like that idea. The applied filter can be a tool-tip, and accessing the filter input box can be through the button. A button is much more discoverable than menu options. I'm thinking two buttons per service might be a bit too much. My thoughts were the settings file should be accessed via command palette e.g. `AWS: open resource explorer filter settings`, or it could also be a button apart of the explorer itself.

**Comment by oscaretu** [comment]
@JadenSimon, I think it would be great the button can have different appearance when there is an active filter.
A button is a more directy accesible that doing right click. If your plan is having only a button, the right click in the service name (Lambda) could add an additional entry to edit the user's setting file, just as you planned using `AWS: open resource explorer filter setting` (so the user can use the one she considers more convenient), if this behaviour doesn't break the UX rules. 
I wasn't able to imagine where you thought  you can put _a button apart of the explorer itself_.

**Comment by JadenSimon** [comment]
We could add a button where it says 'Filter' :
![Screen Shot 2021-07-26 at 12 59 15 PM](https://user-images.githubusercontent.com/31319484/127051260-44e3e098-fccf-4954-bc63-e41a9f09d5a9.png)

**Comment by oscaretu** [comment]
@JadenSimon, I think that position is OK

**Comment by justinmk3** [comment]
> My thoughts were the settings file should be accessed via command palette e.g. `AWS: open resource explorer filter settings`

Storing these filters in the vscode user settings probably makes sense (potentially with the ability to disable this behavior).
